client/src/modules/stock/exit/exit.js

```diff
@@ -117,16 +126,57 @@ function StockExitController(
         headerCellFilter : 'translate',
         cellTemplate : 'modules/stock/exit/templates/expiration.tmpl.html',
       },
-      { field : 'actions', width : 25, cellTemplate : 'modules/stock/exit/templates/actions.tmpl.html' },
+      {
+        field : 'actions',
+        width : 25,
+        cellTemplate : 'modules/stock/exit/templates/actions.tmpl.html',
+      },
     ],
     data : vm.stockForm.store.data,
     fastWatch : true,
     flatEntityAccess : true,
+    showGridFooter : true,
+    gridFooterTemplate,
+    onRegisterApi,
   };
 
   // exposing the grid options to the view
   vm.gridOptions = gridOptions;
 
+  /**
+   * @method exportGrid
+   * @description export the content of the grid to csv.
+   * this function is helpful for handling the export of lot row which are in the grid to a csv file
+   * in a correct format.
+   */
+  vm.exportGrid = () => {
```

Comment:
This case is particular to StockItem object, because each rows of the grid contains a StockItem object, we have to convert these rows in a simple type that ui-grid can easily and correctly export.

This function get columns and rows from the grid, and convert them in a good format (a specific format that we cannot generalize), and export these data with the `uiGridExportService`.

If we have to create a service `X` for that, this service will contains just two line of true process : 
- `uiGridExportService.formatAsCsv(rows, cols)`
- `uiGridExportService.downloadFile()`

Because we have to give columns and rows to this service `X` and this case is not frequent and because of these true line of true process that we can call in the controller, so we decide to do not use a service for that job.

Maybe we can correctly rename this function in something like `formatRowsAndColumns`, and we can add another function `exportGridToCsv` which contains `uiGridExportService.formatAsCsv(rows, cols)` and `uiGridExportService.downloadFile()`

Reply:
It's actually AngularJS best practice to try and make controllers as small as possible.  You can find mention of this in [the AngularJS documentation for controllers](https://docs.angularjs.org/guide/controller).

> In general, a Controller shouldn't try to do too much. It should contain only the business logic needed for a single view.

For more discussion on why this is considered AngularJS best practice, you can [check out this StackOverflow thread](https://stackoverflow.com/questions/31454488/angularjs-why-not-write-logic-in-controller).

Services are the ideal place to put logic, no matter how small they are - it will make it easier for us to test.  For example, see our [VoucherFormTests](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/test/client-unit/services/VoucherForm.spec.js).  Basically, we can test creating a voucher without ever having HTML.

If you really believe this is the best place for it, I'll merge it in.  But it does feel like the extra ~25 lines of code could be abstracted a bit.